controllers/observability/observability_controller.go

```diff
@@ -0,0 +1,71 @@
+package observability
+
+import (
+	"context"
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/rest"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+
+	"github.com/kubevirt/hyperconverged-cluster-operator/pkg/alertmanager"
+)
+
+var (
+	log         = logf.Log.WithName("controller_observability")
+	periodicity = 1 * time.Hour
+)
+
+type Reconciler struct {
+	config *rest.Config
+	events chan event.GenericEvent
+
+	amApi *alertmanager.Api
+}
+
+func (r *Reconciler) Reconcile(_ context.Context, _ ctrl.Request) (ctrl.Result, error) {
+	log.Info("Reconciling Observability")
+
+	if err := r.ensurePodDisruptionBudgetAtLimitIsSilenced(); err != nil {
+		return ctrl.Result{}, err
+	}
+
+	return ctrl.Result{}, nil
+}
+
+func NewReconciler(config *rest.Config) *Reconciler {
+	return &Reconciler{
+		config: config,
+		events: make(chan event.GenericEvent),
+	}
+}
+
+func SetupWithManager(mgr ctrl.Manager) error {
+	log.Info("Setting up controller")
+
+	r := NewReconciler(mgr.GetConfig())
+	r.startEventLoop()
+
+	return ctrl.NewControllerManagedBy(mgr).
+		Named("observability").
+		WatchesRawSource(source.Channel(
+			r.events,
+			&handler.EnqueueRequestForObject{},
+		)).
+		Complete(r)
+}
+
+func (r *Reconciler) startEventLoop() {
+	go func() {
+		for {
+			r.events <- event.GenericEvent{
+				Object: &metav1.PartialObjectMetadata{},
+			}
+			time.Sleep(periodicity)
+		}
```

Comment:
Please consider using the standard library `time.Ticker` instead.

Reply:
Also, please consider to have some escape mechanism; e.g.
for {
	select {
		case <- r.ticker:
			r.events <- event.GenericEvent{
				Object: &metav1.PartialObjectMetadata{},
			}
		case <- something: // (may be a context done channel, or a termination signal channel)
			return
	}
}

Reply:
what about dropping this event event source and simply using something like `return ctrl.Result{Requeue: true, RequeueAfter: periodicity}, nil` in `Reconcile`?

Reply:
Sound like a hack to me.

@machadovilaca, in a second thought, why we even need this logic as a controller? It can be done as a regular go-routine. The only thing we must consider is to make sure the go-routine is still alive, but this is also true for the current go-routine.

Reply:
uhh nice, changed

Reply:
controller-runtime docs recommends to use`RequeueAfter` to poll services that cannot be watched
https://github.com/kubernetes-sigs/controller-runtime/blob/b33709fbf37f66e069c9607888bb6b7ef107b0c3/designs/cache_options.md?plain=1#L140-L147

Reply:
ok, cool, but this whole thing is not k8s resource watching, but a periodic http querying, so I don't think we need the controller mechanism here. It's overkill.